sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -552,6 +523,40 @@ case class DataSource(
         sys.error(s"${providingClass.getCanonicalName} does not allow create table as select.")
     }
   }
+
+  /** Returns an [[InMemoryFileIndex]] that can be used to get partition schema and file list. */
+  private def createInMemoryFileIndex(globbedPaths: Seq[Path]): InMemoryFileIndex = {
```

Comment:
this can be `def createInMemoryFileIndex(checkEmptyGlobPath: Boolean)`

Reply:
and we can merge `checkAndGlobPathIfNecessary` and `createInMemoryFileIndex`

Reply:
No, we can't. In some case we need to check the glob files, while we don't need to create `InMemoryFileIndex`